lua/nvim-treesitter/locals.lua

```diff
@@ -2,20 +2,35 @@
 -- Locals are a generalization of definition and scopes
 -- its the way nvim-treesitter uses to "understand" the code
 local api = vim.api
-local ts = vim.treesitter
 
 local queries = require'nvim-treesitter.query'
 local parsers = require'nvim-treesitter.parsers'
 
-local M = {
-  locals = {}
+local default_dict = {
+  __index = function(table, key)
+    local exists = rawget(table, key)
+    if not exists then
+      table[key] = {}
+    end
+    return rawget(table, key)
+  end
 }
 
-function M.collect_locals(bufnr)
+local query_chache
+if not query_chache then
```

Comment:
wont this module global objects be initialized only on first `require` ?

Reply:
Isn't that the goal of a cache to keep results from previous requires? I wanted to initialize the dict only once and then reuse it.

Reply:
what i meant was that you probably doesn't need to check query_cache exists, as this is the global scope of the module and this might not be initialized twice anyway.